src/core/shutdown.c

```diff
@@ -146,6 +146,9 @@ static int parse_argv(int argc, char *argv[]) {
 }
 
 static int switch_root_initramfs(void) {
+        if (mount(NULL, "/run", NULL, MS_REMOUNT, NULL) < 0)
```

Comment:
hmm, we initially mount /run with MS_NOSUID|MS_NODEV|MS_STRICTATIME, I think we should leave that part in, and not drop that too.

Also this should have a brief comment explaining why, as the reason is really not that obvious.

Reply:
Well, at shutdown, we don't know if systemd has mounted `/run` with `NOSUID|NODEV|MS_STRICTATIME` or if `update-initramfs|dracut` mounted it without those flags. I guess ideal behavior here is "all the same flags, but toggle noexec->exec".

Reply:
Also my thinking here was "this now needs flags as if it is a rootfs, cause we are about to pivot there". Hence, i left out `MS_NOSUID|MS_NODEV|MS_STRICTATIME`, however, shutdown.c is the one defining what the shutdown environment looks like, thus it is also free to dictate what the mount flags for shutdown rootfs should be.

Reply:
I am a bit confused. man 2 mount says that you cannot change MS_NOEXEC (anymore) with MS_REMOUNT. Did you make sure noexec really was removed?

Also, if we do want to mount /run noexec, then it's probably better/safer to mount a separate ramfs instance in /run/initramfs instead of trying a remount at shutdown.

Reply:
@medhefgo tested locally, it does drop noexec... at least in my manpage it says:

```
Remounting an existing mount
The following mountflags can be changed: MS_LAZYTIME, MS_MANDLOCK, MS_NOATIME, MS_NODEV, MS_NODIRATIME, MS_NOEXEC, MS_NOSUID, MS_RELATIME, MS_RDONLY, and MS_SYNCHRONOUS.  Attempts to change the setting of the MS_DIRSYNC flag during a remount are silently ignored.
```

So it seems to be all good.

Well, a separate `/run/initramfs`... possibly, but then the initramfs-tools need to set that up. And if systemd mounts that, we somehow need to preserve existing content that startup initrd may leave there. Cause if those things are dropped, it would affect the shutdown initrd.

Reply:
Yeah, nvm. The manpage on the net I was reading was super out of date I guess.